vscode-lean4/test/suite/utils/helpers.ts

```diff
@@ -4,6 +4,87 @@ import { privateEncrypt } from 'crypto';
 import * as vscode from 'vscode';
 import { InfoProvider } from '../../../src/infoview';
 import { LeanClient} from '../../../src/leanclient';
+import * as ps from 'ps-node';
+
+export async function findProcs(name: string) : Promise<ps.Program[]> {
+  // A simple pid lookup
+  return await new Promise<ps.Program[]>((resolve) => {
+    ps.lookup({ command: name }, function(err, resultList ) {
+      if (err) {
+        resolve([]);
+      }
+      resolve(resultList);
```

Comment:
```suggestion
      } else {
        resolve(resultList);
      }
```
I just read the relevant part of the ECMAScript spec and the semantics is indeed the same, but calling resolve only once seems infinitely clearer to me.